appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java

```diff
@@ -165,26 +166,23 @@ private GoalResourceAssembler createResourceAssembler(UUID requestingUserId, UUI
 		return new GoalResourceAssembler(requestingUserId, userId);
 	}
 
-	public static Resources<GoalDto> createAllGoalsCollectionResource(UUID requestingUserId, UUID userId,
+	public static CollectionModel<GoalDto> createAllGoalsCollectionResource(UUID requestingUserId, UUID userId,
 			Set<GoalDto> allGoalsOfUser)
 	{
-		return new Resources<>(new GoalResourceAssembler(requestingUserId, userId).toResources(allGoalsOfUser),
+		return new CollectionModel<>(new GoalResourceAssembler(requestingUserId, userId).toCollectionModel(allGoalsOfUser),
 				getAllGoalsLinkBuilder(requestingUserId, userId).withSelfRel());
 	}
 
-	public static ControllerLinkBuilder getAllGoalsLinkBuilder(UUID requestingUserId, UUID userId)
+	public static WebMvcLinkBuilder getAllGoalsLinkBuilder(UUID requestingUserId, UUID userId)
 	{
 		GoalController methodOn = methodOn(GoalController.class);
 		return linkTo(methodOn.getAllGoals(null, requestingUserId.toString(), userId));
 	}
 
 	private void setActivityCategoryId(GoalDto goal)
 	{
-		Link activityCategoryLink = goal.getLink(curieProvider.getNamespacedRelFor(ACTIVITY_CATEGORY_REL));
-		if (activityCategoryLink == null)
-		{
-			throw InvalidDataException.missingActivityCategoryLink();
-		}
+		Link activityCategoryLink = goal.getLink(curieProvider.getNamespacedRelFor(ACTIVITY_CATEGORY_REL))
+				.orElseThrow(() -> InvalidDataException.missingActivityCategoryLink());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1612)